### PR TITLE
Update readme to include missing examples and instructions

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
Signed-off-by: Yury Fridlyand <yuryf@bitquilltech.com>

See readme: https://github.com/Bit-Quill/opensearch-net/blob/update-readme/README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,13 +42,114 @@ Include OSC in your .csproj file.
...
+    Message = "Trying out OSC, so far so good?"
+};
+
+var indexingResponse = client.Index(tweet, idx => idx.Index("mytweetindex")); //or specify index via settings.DefaultIndex("mytweetindex");
+```
+
+All the calls have async variants:
+
+```csharp
+var indexingResponse = client.IndexAsync(tweet, idx => idx.Index("mytweetindex")); // returns a Task<IndexResponse>
```
Comment (MaxKsyunz) [inline README.md]: Shouldn't this be .Index instead of IndexAsync?

File: README.md [hunk 2]
```diff
@@ -42,13 +42,114 @@ Include OSC in your .csproj file.
...
+    .From(0)
+    .Size(10)
+    .Query(q => q
+        .Term(t => t.User, "kimchy") || q
+        .Match(mq => mq.Field(f => f.User).Query("osc"))
+    )
+);
+```
+
+As well as an object initializer syntax if lambdas aren't your thing:
```
Comment (MaxKsyunz) [inline README.md]: I'd remove or reword "if lambdas aren't your thing" -- it's too informal.

**Review by alexmeizer — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -42,13 +42,114 @@ Include OSC in your .csproj file.
...
+};
+
+var searchResponse = client.Search<Tweet>(request);
+```
+### Falling back to OpenSearch.Net
+
+OSC also includes and exposes the low-level [OpenSearch.Net](https://github.com/opensearch-project/opensearch-net/tree/main/src/OpenSearch.Net) client that you can fall back to in case anything is missing:
+
+```csharp
+//.LowLevel is of type IOpenSearchLowLevelClient
```
Comment (MaxKsyunz) [inline README.md]: I like that we mention the type of intermediate properties. 

It'd be more clear to use OpenSearchClient.LowLevel to refer to LowLevel property in the comments.

Same for the Search method -- IOpenSearchLowLevelClient.Search.

File: README.md [hunk 4]
```diff
@@ -42,13 +42,114 @@ Include OSC in your .csproj file.
...
+				value= "OSC"
+			}
+		}
+	}
+}));
+```
+
 # [OpenSearch.Net](src/OpenSearch.Net)
 
 A low-level, dependency free client that has no opinions how you build and represent your requests and responses.
```
Comment (MaxKsyunz) [inline README.md]: I'd reword this as well to avoid "no opinions." 

Is OpenSearch.Net a just a .Net wrapper for the REST API? I'd say so here.

**Comment by MaxKsyunz** [comment]
The sample code should use packages from a NuGet repository. It's odd otherwise that the sample code does not use the packages we release.

There is no need to explain how to connect to a NuGet repository -- nuget.org is set up by default in any .Net dev environment and that's where our packages will be..

**Review by MaxKsyunz — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Yury-Fridlyand) [reply README.md]: No, because async implementation is mentioned here.

File: README.md (hunk 2 above)
Comment (Yury-Fridlyand) [reply README.md]: Thank you, fixed in 162693602.

File: README.md (hunk 1 above)
Comment (Yury-Fridlyand) [reply README.md]: See update in 162693602.

File: README.md (hunk 4 above)
Comment (Yury-Fridlyand) [reply README.md]: Thank you, fixed in 162693602.

File: README.md (hunk 3 above)
Comment (Yury-Fridlyand) [reply README.md]: See update in 162693602.

**Review by ghost — APPROVED** [review]

**Review by MaxKsyunz — APPROVED** [review]

**Review by Yury-Fridlyand — COMMENTED** [review]
LGFM

**Review by alexmeizer — APPROVED** [review]
